### PR TITLE
Add documentation

### PR DESCRIPTION
**Pull request by khandelwal**
This adds some documentation to various functions that have been written. 

This builds on a previous branch (and hence contains all those changes), but once that's merged - this will look cleaner.

### CODE REVIEW

File: docusearch/management/commands/import_documents.py [hunk 1]
```diff
@@ -51,6 +57,9 @@ def convert_to_text(file_name, doc_path):
 
 
 def create_document(document, release_slug):
+    """ Create a Document object representing the document. This also uploads
```
Comment (geramirez) [inline docusearch/management/commands/import_documents.py:60]: When I looked at this function the first time around I had to a bit of searching to figure out how the documents were uploaded to s3. Maybe add a bit about `django-storages` here or in the model.

File: docusearch/search_indexes.py [hunk 2]
```diff
@@ -4,8 +4,14 @@
 
 
 class DocumentIndex(indexes.SearchIndex, indexes.Indexable):
+    """ The Haystack related class that defined what's going to be indexed. """
 
+    # This uses a template, but essentially indexes the extracted text of a 
+    # document. 
     text = indexes.CharField(document=True, use_template=True)
```
Comment (geramirez) [inline docusearch/search_indexes.py:11]: Should we be dropping in `title` into the template too?

Comment (khandelwal) [inline docusearch/search_indexes.py:11]: Let's leave that for a different pull request.

File: docusearch/management/commands/import_documents.py [hunk 3]
```diff
@@ -65,10 +74,14 @@ def create_document(document, release_slug):
     doc_file = File(open(doc_path, 'rb'))
     filename = os.path.basename(doc_path)
 
+    # On save() django-storages uploads this file to S3
```
Comment (khandelwal) [inline docusearch/management/commands/import_documents.py:77]: @ramirezg I added this line to help indicate where the upload happens.
